### PR TITLE
zh, en: add 2.0.4 release note

### PR DESCRIPTION
**Pull request by GMHDBJD**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-dm) that's required for repo owners to accept my contribution.

### Which DM version(s) do your changes apply to? (Required)

- [x] master (the latest development version, including v3.0 changes)
- [x] v2.0 (TiDB DM 2.0 versions)
- [ ] v1.0 (TiDB DM 1.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by GMHDBJD** [comment]
/needs-cherry-pick-release-2.0

**Review by lance6716 — APPROVED** [review]

**Comment by lance6716** [comment]
/hold

wait CI/CD

File: en/releases/2.0.4.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+DM version: 2.0.4
+
+## Improvements
+
+- Support rescheduling and automatically resuming tasks after a DM-worker goes offline first and then comes back online during the full import [#1784](https://github.com/pingcap/dm/pull/1784)
+- Add the metric `replicationLagGauge` to monitor replication delay [#1759](https://github.com/pingcap/dm/pull/1759)
+- Restore schemas in parallel during the full import [#1701](https://github.com/pingcap/dm/pull/1701)
+- Support automatically adjusting the time_zone settings of both the upstream and downstream databases [#1714](https://github.com/pingcap/dm/pull/1714)
+- Improve the speed of rolling back incremental replication tasks after the tasks meet errors  [#1705](https://github.com/pingcap/dm/pull/1705)
```
Comment (CharLotteiu) [inline en/releases/2.0.4.md:17]:
```suggestion
- Improve the speed of rolling back incremental replication tasks after the tasks meet errors [#1705](https://github.com/pingcap/dm/pull/1705)
```

File: en/releases/2.0.4.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+- Automatically adjust GTID according to checkpoints when GTID is enabled during the incremental replication [#1745](https://github.com/pingcap/dm/pull/1745)
+- Detect the versions of upstream and downstream databases and record the versions in log files [#1693](https://github.com/pingcap/dm/pull/1693)
+- Use the schema from the dump stage of the full export as the initial schema for the incremental replication task of the same data source [#1754](https://github.com/pingcap/dm/pull/1754)
+- Decrease the time that the safe mode lasts after the incremental task is restarted to 1 minute to improve the replication speed [#1779](https://github.com/pingcap/dm/pull/1779)
```
Comment (CharLotteiu) [inline en/releases/2.0.4.md:21]:
```suggestion
- Decrease the time that the safe mode lasts after the incremental task is restarted to one minute to improve the replication speed [#1779](https://github.com/pingcap/dm/pull/1779)
```

**Review by CharLotteiu — APPROVED** [review]
rest LGTM

**Pull request merged**